components/apimgt/org.wso2.carbon.apimgt.rest.api/src/main/java/org/wso2/carbon/apimgt/rest/api/validation/ParameterEqualityValidator.java

```diff
@@ -0,0 +1,86 @@
+ *  limitations under the License.
+ */
+
+package org.wso2.carbon.apimgt.rest.api.validation;
+
+import org.apache.commons.beanutils.BeanUtils;
+import org.apache.commons.lang.StringUtils;
+import org.wso2.carbon.apimgt.rest.api.exception.InternalServerErrorException;
+import org.wso2.carbon.apimgt.rest.api.validation.constraints.ValidateParamEquality;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import javax.validation.constraintvalidation.SupportedValidationTarget;
+import javax.validation.constraintvalidation.ValidationTarget;
+import java.lang.reflect.InvocationTargetException;
+
+@SupportedValidationTarget(ValidationTarget.PARAMETERS)
+public class ParameterEqualityValidator implements ConstraintValidator<ValidateParamEquality, Object[]> {
+
+    private int firstParamIndex;
+    private int secondParamIndex;
+    private String firstField;
+    private String secondField;
+    private String errorMessage;
+
+    @Override
+    public void initialize(ValidateParamEquality validateEquality) {
+        firstParamIndex = validateEquality.index0();
+        secondParamIndex = validateEquality.index1();
+        firstField = validateEquality.index0Field();
+        secondField = validateEquality.index1Field();
+        errorMessage = validateEquality.message();
+    }
+
+    @Override
+    public boolean isValid(Object object[], ConstraintValidatorContext constraintValidatorContext) {
+
+        try {
+            Object firstObj = object[firstParamIndex];
+            Object secondObj = object[secondParamIndex];
+
+            if (firstField != null && !StringUtils.isEmpty(firstField)) {
+                firstObj = BeanUtils.getProperty(firstObj,
+                        firstField); //if first field is present, take first object as the field value of the firstObj
+            }
+
+            if (secondField != null && !StringUtils.isEmpty(secondField)) {
+                secondObj = BeanUtils.getProperty(secondObj,
+                        secondField); //if second field is present, take second object as the field value of the secondObj
+            }
+
+            //let @NotNull for checking null values
+            if (firstObj == null || secondObj == null) {
+                return true;
+            }
+
+            boolean isValid = firstObj.equals(secondObj);
+
+            if (!isValid && ValidateParamEquality.DEFAULT_ERROR_MESSAGE.equals(errorMessage)) {
+                constraintValidatorContext.disableDefaultConstraintViolation();
+                constraintValidatorContext.buildConstraintViolationWithTemplate(
+                        "Values \\{" + firstObj + "," + secondObj + "\\} must be equal").addConstraintViolation();
+            }
+
+            return isValid;
+
+        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException e) {
+            throw new InternalServerErrorException(e);
```

Comment:
Fixed in #1528 (https://github.com/wso2/carbon-apimgt/commit/40bb7af6f0d8c412f6e6f80497a52e8d88ff7b9b)